### DISCUSSION TITLE
Problem with Grouped CollectionView on Windows

### DISCUSSION THREAD

**Discussion by XamerDev** [dotnet/maui#7528] [Q&A] [answered]
Hello I need to create grouped collectionView but I'm facing some issues on Windows. On android works well
Code

```
public class FoodGroup : ObservableCollection<Food>
        {
            public string Name { get; private set; }

            public FoodGroup(string name, ObservableCollection<Food> foods) : base(foods)
            {
                Name = name;
            }
        }
        public ObservableCollection<FoodGroup> Lines { get; private set; } = new ObservableCollection<FoodGroup>();

```
and when I trying add something to Lines collection:
`Lines.Add(new FoodGroup("group1", new ObservableCollection<Food> {  }));`
I facing this issue:
"MauiApp3.ViewModels.JournalViewModel+FoodGroup" is not of type "MauiApp3.Models.Food" and cannot be used in this generic collection. Arg_ParamName_Name”

Issue appears only on Windows, on android emulator everything is working properly.
Is it something connecting with MAUI itself or it's just my misunderstanding of something?
Thank you

**Comment by hartez** [comment, marked as answer]
Sounds like a bug in the Windows implementation. Going to convert this to an issue so we can take a look.

**Reply by jari-schuurman** [reply]
@hartez any update?

**Comment by rkeyuutel** [comment]
I get this error when using a CollectionView.GroupFooterTemplate.

CollectionView.GroupHeaderTemplate seems fine.
